solr/core/src/java/org/apache/solr/core/SolrCore.java

```diff
@@ -2062,6 +2064,14 @@ public FacetParserFactory getFacetParserFactory(String name) {
    * RequestHandlers need access to the updateHandler so they can all talk to the same RAM indexer.
    */
   public UpdateHandler getUpdateHandler() {
+    // hitesh
+    if (!isVersionBucketSeeded.get()) {
+      synchronized (isVersionBucketSeeded) {
+        seedVersionBuckets();
+        isVersionBucketSeeded.set(true);
+      }
+    }
+
```

Comment:
is `seedVersionBuckets()` idempotent? It could be called more than once here, by different threads.

Reply:
`seedVersionBuckets` version is initialized in solrcore constructor. Thus I just put here to initialized once only. 
I think it(seedVersionBuckets) called after core-recovery as well.

Need to bit understand its usage to find the right place for lazy loading.